### ISSUE TITLE
Workspace root configuration not working in Windows

### ISSUE THREAD

**Issue by kuldeeparora89** [eclipse-emfcloud/emfcloud-modelserver-theia#47]
Hi,

I am no able to configure workspace when running theia in windows env with model server.

Issue -
I have a theia workspace at below path -
/d:/kuldeep/lab/theia-wksp
Theia client encode this url & send it to the model server backend as below for configuration -
/d%3A/kuldeep/lab/theia-wksp

DefaultUriHelper convert this to URI , which is causing the issue -
```
public static String withoutFileScheme(final String uriString) {
      URI uri = uriString.startsWith("file:") ? URI.createURI(uriString) : URI.createFileURI(uriString);
      return withoutFileScheme(uri).toString();
   }
```
This changes url to-
/d%253A/kuldeep/lab/theia-wksp

Due to this URL is not getting decoded correctly.

As a workaround, I have extended the DefaultUriHelper to override the below method-
	

             @Override
	       public Optional<URI> toFileUri(String fileUrl) {
		// TODO Auto-generated method stub
		       try {
			return super.toFileUri(URLDecoder.decode(fileUrl, "UTF-8"));

		} catch (NullPointerException | IllegalArgumentException | UnsupportedEncodingException e) {
			LOG.warn(String.format("Could not convert to filePath! â€™%sâ€™ is not a valid URL", fileUrl));
			return Optional.empty();
		}
	    }
          

Basically decoding the URL before converting it to URI.

**Comment by martin-fleck-at** [comment]
@kuldeeparora89 Thank you for raising this issue! We already had some issues with Windows URIs once but it seems that there are still some cases where such URIs cause problems. We'll probably need to double-check all usages and occurrences again and maybe centralize these methods.

**Comment by martin-fleck-at** [comment]
Closed with https://github.com/eclipse-emfcloud/emfcloud-modelserver/pull/142. If you encounter any more problems, please let us know.
